### PR TITLE
XRandR Improvements

### PR DESCRIPTION
**Pull request by scymtym**
* Tab, whitespace, comment  cleanup

* Separate package for the extension

* Fix and complete various things

  * Notify events

  * Mask and type definition macros

  * Accessor definitions for `panning`, `mode-info`, etc

  * A few incomplete definitions of CRTC and output functions

### CODE REVIEW

**Comment by JMC-design** [comment]
Does the rr-panning-put really still not work?  I was kinda hoping to see what the issue was with that.

File: extensions/randr.lisp [hunk 1]
```diff
@@ -15,298 +16,384 @@
...
-
-(deftype mode-flag-mask ()
-  '(or mask32 (clx-list mode-flag-mask-class)))
-
-  ;; temporarily here since not in xrender.lisp
+  (defconstant +connection+
+    #(:connected :disconnected :unknown-connection))
 
+  ;; TODO (deftype connection () '(or +connected+ +disconnected+ +unknown-connection+))
+  )
```
Comment (dkochmanski) [inline extensions/randr.lisp:220]: s/or/member/?

File: extensions/randr.lisp [hunk 2]
```diff
@@ -563,538 +539,639 @@ include, if it can, the current refresh rate and the refresh rate
...
-;;                                      ;  (card32-get 8) ; new timestamp
-;;      )))
+       (sequence-get :result-type result-type :length cur-num-params :index cur-param-start)))))
+
+(defun set-crtc-transform (display crtc transform
+                           &key filter-name filter-parameters)
+  "FIXME:Transform may be a list or vector of length 9.  ?perhaps allow length 6?"
+  (declare (ignore display crtc transform filter-name filter-parameters))
+  (error "not implemented")
+  #+TODO (let ((seq          (if filter-parameters (coerce filter-parameters 'vector) nil ))
```
Comment (dkochmanski) [inline extensions/randr.lisp:1058]: please use `#+(or)` or comment out the block.

File: extensions/randr.lisp [hunk 3]
```diff
@@ -563,538 +539,639 @@ include, if it can, the current refresh rate and the refresh rate
...
-
-;;     (values
-;;      (member8-vector-get 1 +rr-config-status+)
-;;                                      ;  (card32-get 8) ; new timestamp
-;;      )))
+       (sequence-get :result-type result-type :length cur-num-params :index cur-param-start)))))
+
+(defun set-crtc-transform (display crtc transform
+                           &key filter-name filter-parameters)
+  "FIXME:Transform may be a list or vector of length 9.  ?perhaps allow length 6?"
```
Comment (dkochmanski) [inline extensions/randr.lisp:1055]: this is a comment, not a documentation string - please make it so

File: extensions/randr.lisp [hunk 4]
```diff
@@ -563,538 +539,639 @@ include, if it can, the current refresh rate and the refresh rate
...
+             (data +rr-setcrtctransform+)
+             (card32 crtc)
+             (card16 param-length)
+             (pad16)
+             (transform transform)
+             (card16 name-length)
+             (pad16)
+             (string filter-name)
+             ;; ((sequence :format card32) seq)
+                                        ;((sequence :format card32) seq)
```
Comment (dkochmanski) [inline extensions/randr.lisp:1075]: fix indentation and uniform comment

**Review by dkochmanski — CHANGES_REQUESTED** [review]
only minor remarks

**Comment by scymtym** [comment]
Sorry, this was meant to be a draft, not ready for review. Since I promised @JMC-design to let them have a look.

**Comment by dkochmanski** [comment]
OK, thanks. I've converted PR to draft.

**Comment by scymtym** [comment]
One issue I couldn't come to a decision on: should the synchronous operations that receive a success or failure indication in the reply just return that value or signal a condition on failure (example: `get-output-info` returns either `:success` or `:failure` as the first return value). I lean towards just returning the value and leave the construction of convenience layers to clients but that does, of course, make the base interface less convenient.

Similarly, some operations return "meta data" in the form of timestamps and "configuration timestamps" along with the primary data. The order of return values is currently mostly modeled after the order of fields in the respective XRandR reply structure. However, it might make sense to return the meta data values as the final values since they seem to not always be needed.

File: extensions/randr.lisp [hunk 5]
```diff
@@ -16,72 +17,143 @@
 ;;; but WITHOUT ANY WARRANTY; without even the implied warranty of
 ;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 
-(in-package :xlib)
```
Comment (dkochmanski) [inline extensions/randr.lisp:20]: this change seems to be against rules established in the codebase, all extensions are defined in the xlib package. macros also assume that everything is defined in the xlib package. In light of that please define the extension in the same package with the rr prefix like it was before. I agree that this is not optimal.

File: extensions/randr.lisp [hunk 6]
```diff
@@ -159,213 +159,254 @@
...
 (define-extension "RANDR"
   :events (:rr-screen-change-notify
-           :rr-crtc-change-notify
-           :rr-output-change-notify
-           :rr-output-property-notify)
+           :rr-notify
+           ; :rr-crtc-change-notify
+           ; :rr-output-change-notify
+           ; :rr-output-property-notify
+           )
```
Comment (dkochmanski) [inline extensions/randr.lisp:172]: please elaborate in the commit message why these events are commented out and elaborate what do you mean by "fix various notify events".

File: extensions/randr.lisp [hunk 7]
```diff
@@ -1157,27 +1157,38 @@ configuration, and does not poll for hardware changes."
...
+
+Note that successful execution of this request may set but not apply
+the new transform immediately. Calling `set-crtc-config' (possibly
+with an unchanged configuration) can be used to force the change into
+effect."
+  (declare (type display                       display)
+           (type crtc-id                       crtc)
+           (type (vector (unsigned-byte 16) 9) transform)
+           (type string                        filter-name))
+  (when filter-parameters (error "Filter parameters are not currently supported"))
```
Comment (dkochmanski) [inline extensions/randr.lisp:1185]: please break a line here

**Review by dkochmanski — CHANGES_REQUESTED** [review]

**Pull request merged**